oshmem/proc/proc.h

```diff
@@ -35,30 +37,15 @@ struct oshmem_group_t;
 
 #define OSHMEM_PE_INVALID   (-1)
 
-/**
- * Remote Open SHMEM process structure
- *
- * Remote Open SHMEM process structure.  Each process contains exactly
- * one oshmem_proc_t structure for each remote process it knows about.
- */
-struct oshmem_proc_t {
-    opal_proc_t                    super;
-    /* endpoint data */
-    void *proc_endpoints[OMPI_PROC_ENDPOINT_TAG_MAX];
-    /*
-     * All transport channels are globally ordered.
-     * pe(s) can talk to each other via subset of transports
-     * these holds indexes of each transport into global array
-     *  proc -> id, where id can be btl id in yoda or mxm ptl id
-     *  in ikrit
-     *  spml is supposed to fill this during add_procs()
-     **/
-    int                             num_transports;
-    char                           *transport_ids;
+struct oshmem_proc_data_t {
+    int num_transports;
+    char * transport_ids;
```

Comment:
AFAIK, we (officially) don't care about this case.  Specifically: I don't think we make any strong attempts to preserve internal ABIs even between subsequent Z releases of vX.Y.Z.

But we can certainly try to be social to 3rd party component authors.  Meaning: perhaps we can flip the members on master, but keep them in the same order on v2.x.